### ISSUE TITLE
Progressbar display loss nan.

### ISSUE THREAD

**Issue by sooftware** [PyTorchLightning/pytorch-lightning#7672]
## Progress bar display loss nan.
  
When start training, progress bar display `loss: nan`. I tested more than 10 models but, all model's loss displayed `nan`.   
I made a test file and took a log to see if the loss was really nan, and the loss was displayed normally here.  
     
Below are some of the areas that I have implemented that may be relevant to this. If there is anything strange, please give me feedback.

```python
class Model(pl.LightningModule):
    def configure_optimizers(self) -> Dict[str, Union[torch.optim.Optimizer, object, str]]:
        r"""
        Configure optimizer for training.

        Returns:
            dict: dictionary contains `optimizer`, `scheduler`, `monitor`
        """
        SUPPORTED_OPTIMIZERS = {
            "adam": Adam,
            "adamp": AdamP,
            "radam": RAdam,
            "adagrad": Adagrad,
            "adadelta": Adadelta,
            "adamax": Adamax,
            "adamw": AdamW,
            "sgd": SGD,
            "asgd": ASGD,
            "novograd": Novograd,
        }
        assert self.configs.model.optimizer in SUPPORTED_OPTIMIZERS.keys(), \
            f"Unsupported Optimizer: {self.configs.model.optimizer}\n" \
            f"Supported Optimizers: {SUPPORTED_OPTIMIZERS.keys()}"

        self.optimizer = SUPPORTED_OPTIMIZERS[self.configs.model.optimizer](
            self.parameters(),
            lr=self.configs.lr_scheduler.lr,
        )
        self.scheduler = SCHEDULER_REGISTRY[self.configs.lr_scheduler.scheduler_name](
            optimizer=self.optimizer,
            configs=self.configs,
        )

        return {
            'optimizer': self.optimizer,
            'scheduler': self.scheduler,
            'monitor': 'metric_to_track',
        }

    def optimizer_step(
            self,
            epoch=None,
            batch_idx=None,
            optimizer=None,
            optimizer_idx=None,
            optimizer_closure=None,
            on_tpu=None,
            using_native_amp=None,
            using_lbfgs=None,
    ) -> None:
        self.optimizer.step()
        self.optimizer.zero_grad()
        if (isinstance(self.scheduler, ReduceLROnPlateauScheduler)
            or isinstance(self.scheduler, WarmupReduceLROnPlateauScheduler)) \
                and self.trainer.global_step % self.trainer.val_check_interval == 0:
            self.scheduler.step(self.current_val_loss)
        else:
            self.scheduler.step()

    def configure_criterion(self, criterion_name: str) -> nn.Module:
         ...

```

**Comment by ethanwharris** [comment]
Hi @sooftware, thanks for the issue :smiley: Could you provide a reproducible script, ideally using our [**BoringModel**](https://colab.research.google.com/drive/1HvWVVTK8j2Nj52qU4Q4YCyzOm0_aLQF3?usp=sharing), showing the bug in action? Without that it will be difficult for us to assist you.
